vtex/faststore
# 404 status code on notFound page

## What's the purpose of this pull request?
Currently, we use the client-side redirect approach to our notFound page so Google bot knows the initial path was a 404. the only missing part of the puzzle is to make the server return a 404 status code for this page. This PR does that and fixes eventual issues in the nginx config generator so we also have this behavior in our platform

## How it works? 
It creates a notFound page and than it creates a redirect from /404 to this page with 404 status code. This way we make netlify and our nginx work correctly

## How to test it?
Enter an unknown product, like /blah/p and check, in devTool's network that you receive a 404 status code

[marinbrasil](https://github.com/vtex-sites/marinbrasil.store/pull/281)
[storecomponents](https://github.com/vtex-sites/storecomponents.store/pull/414)

## References
<em>Spread the knowledge: is there any content you used to create this PR that is worth sharing?</em>  

<em>Extra tip: adding references to related issues or mentioning people important to this PR may be good for the documentation and reviewing process</em>
